keps/sig-scalability/0001-enhance-network-performance-benchmark-testcases.md

```diff
@@ -0,0 +1,152 @@
+owning-sig: sig-scalability
+participating-sigs:
+- sig-networks
+- sig-testing
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: 
+creation-date: 2020-08-24
+last-updated: 2020-08-25
+status: implemented
+---
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-goals](#non-goals)
+  - [Challenges and Open Questions [optional]](#challenges-and-open-questions-optional)
+- [Proposal](#proposal)
+  - [Test Scenarios](#test-scenarios)
+  - [Tools](#tools)
+  - [Execution Environment](#execution-environment)
+  - [Future Enhancements](#future-enhancements)
+  - [Benefits](#benefits)
+- [Graduation Criteria](#graduation-criteria)
+  - [Beta -&gt; GA](#beta---ga)
+  - [GA -&gt; Future](#ga---future)
+- [Implementation History](#implementation-history)
+- [Drawbacks [optional]](#drawbacks-optional)
+- [Alternatives [optional]](#alternatives-optional)
+<!-- /toc -->
+
+## Summary
+
+We are proposing to add more testcases for network performance benchmarking.
+
+## Motivation
+
+As Kubernetes is increasingly being adopted by various telco cloud providers for deployments, 
+it brings additional networking requirements/scenarios to be supported.  
+With wide array of network functions being containerized, the need to analyse and 
+monitor the networking performance metrics such as latency, throughput etc has arisen.
+
+We have come across 2 test frameworks which have limited testcases for network performance benchmarking.
+
+a) E2E (test\e2e\network\networking_perf.go): This framework has only 2 testcases for measuring throughput.
+
+b) netperf (https://github.com/kubernetes/perf-tests/tree/master/network/benchmarks/netperf) : This framework 
+   measures only the throughput between client-server pods for various MSS(Maximum Segment size). 
+   Other performance metrics such as latency , loss-rate , response time are not measured.
+
+### Goals
+
+The primary goal of this proposal is to enhance the testcases to cover various 
+networking scenarios :
+
+a)  IPv4 , IPv6 , Dual-Stack.
+b)  Protocols: TCP, UDP, HTTP/HTTPS.
+c)  With services such as ClusterIP , NodePort etc.
+
+We intend to submit tests to measure the following network performance metrics for various 
+scenarios such as different services (ClusterIP/NodePort), with IPv4/IPv6 configured etc.
+
+a)  Throughput (For TCP traffic)
+b)  Loss rate (For UDP traffic)
+c)  Latency/Jitter (For UDP/HTTP/HTTPS)
+d)  Response time (For HTTP/HTTPs)
+
+Our important goal is to make sure these benchmarking testcases are run periodically,
+so that any degradation in performance is detected in early phase of feature delivery.
+
+### Non-goals
+
+Benchmarking and measurements with intermediate services like ingress controllers, load-balancers 
+are out of scope of this proposal.
+
+### Challenges and Open Questions [optional]
+
+1) Which is the suitable hosting framework for proposed tests ? 
+2) Periodic execution of these proposed tests ?
```

Comment:
They should be set up as any other K8s CI/CD job, i.e. using prow. SIG Scalability can provide resources for running these tests. See examples of scalability prow jobs in https://github.com/kubernetes/test-infra/tree/master/config/jobs/kubernetes/sig-scalability

Reply:
Ok will take a look at it.

Reply:
@mm4tt we have updated this spec with HLD(High Level Design) assuming we are going with clusterloader framework. Could you please review it ?